yeat/cli.py

```diff
@@ -8,39 +8,54 @@
 # -------------------------------------------------------------------------------------------------
 
 import argparse
+from argparse import Action
+import json
 from pathlib import Path
 from pkg_resources import resource_filename
 from snakemake import snakemake
+import sys
 import yeat
+from yeat.config import AssemblerConfig
 
 
-ASSEMBLY_ALGORITHMS = ["spades", "megahit", "unicycler"]
+CONFIG_TEMPLATE = [
+    dict(
+        algorithm="spades",
+        extra_args="--meta",
+    ),
+    dict(
+        algorithm="megahit",
+        extra_args="--min-count 5 --min-contig-len 300",
+    ),
+]
 
 
-def check_assemblers(assemblers):
-    algorithms = []
-    for algorithm in assemblers:
-        if algorithm not in ASSEMBLY_ALGORITHMS:
-            message = (
-                f"Found unsupported assembly algorithm with `--assemblers` flag: [[{algorithm}]]!"
-            )
-            raise ValueError(message)
-        if algorithm in algorithms:
-            message = (
-                f"Found duplicate assembly algorithm with `--assemblers` flag: [[{algorithm}]]!"
-            )
-            raise ValueError(message)
-        algorithms.append(algorithm)
+class InitAction(Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        json.dump(CONFIG_TEMPLATE, sys.stdout, indent=4)
+        print()
+        raise SystemExit()
 
 
-def run(read1, read2, assemblers, outdir=".", cores=1, sample="sample", dryrun="dry"):
+def run(read1, read2, algorithms, outdir=".", cores=1, sample="sample", dryrun="dry"):
     snakefile = resource_filename("yeat", "Snakefile")
     r1 = Path(read1).resolve()
+    if not r1.is_file():
+        raise FileNotFoundError(f"No such file: '{r1}'")
     r2 = Path(read2).resolve()
+    if not r2.is_file():
+        raise FileNotFoundError(f"No such file: '{r2}'")
+    assemblers = []
+    extraargs = {}
+    for algorithm in algorithms:
+        assembler = algorithm.algorithm
+        assemblers.append(assembler)
+        extraargs[assembler] = algorithm.extra_args
```

Comment:
For what it's worth, these six lines could be collapsed into two arguably clearer lines using list and dict comprehensions.

```python
assemblers = [config.algorithm for config in assembly_configs]
extra_args = {config.algorithm: config.extra_args for config in assembly_configs}
```